### PR TITLE
tx fee denom whiltelist + PFM ViperStrike patch

### PR DESCRIPTION
**Pull request by ajeet97**
## 1. Overview

- fee denom whitelist `XPRT` `ATOM` `PSTAKE` `OSMO` `stkATOM` `stkOSMO`
- pfm patch

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -340,6 +342,18 @@ func (app *Application) GetChainBondDenom() string {
...
+func (app *Application) GetFeeDenomsWhitelist() []string {
+	chainID := app.ChainID()
+
+	if strings.HasPrefix(chainID, "core-") {
+		return FeeDenomsWhitelistMainnet
+	} else if strings.HasPrefix(chainID, "test-core-") {
+		return FeeDenomsWhitelistTestnet
+	}
+
+	return []string{"stake"}
```
Comment (puneet2019) [inline app/app.go:354]: can you for random chain do something like allow all?  special condition? Not for this release but in general..

Comment (ajeet97) [reply app/app.go:354]: sure 👍

File: app/fee_denom_whitelist_decorator.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		panic("at least one fee denom must be whitelisted")
+	}
+
+	whitelistMap := map[string]struct{}{}
+	for _, denom := range denomsWhitelist {
+		// must be valid denom
+		if err := sdk.ValidateDenom(denom); err != nil {
+			panic(fmt.Sprintf("invalid denoms whiltelist; err: %v", err))
+		}
+		whitelistMap[denom] = struct{}{}
```
Comment (puneet2019) [inline app/fee_denom_whitelist_decorator.go:29]: ok could just do map[string]bool and set it to true

Comment (puneet2019) [reply app/fee_denom_whitelist_decorator.go:29]: its cool then..

**Review by xlab — APPROVED** [review]

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**